tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,587 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+a : Bool
+a = (l |> (not List.isEmpty)) 
+"""
+                        ]
+        , test "should replace 0 /= (l |> List.length) with (not List.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= (l |> List.length)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not List.isEmpty)"
+                            , details =
+                                [ "List.length takes as long to run as the list is long"
+                                , "whereas List.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= (l |> List.length)"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a =  (l |> (not List.isEmpty))
+"""
+                        ]
+        ]
+
+
+setTests : Test
+setTests =
+    describe "Set.size should be Set.isEmpty"
+        [ test "should replace Set.size s == 0 with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = Set.size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Set.size takes as long to run as the list is long"
+                                , "whereas Set.isEmpty runs in constant time."
+                                ]
+                            , under = "Set.size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = Set.isEmpty s 
+"""
+                        ]
+        , Test.skip <|
+            test "should replace size s == 0 with isEmpty s" <|
+                \() ->
+                    """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = size s == 0
+"""
+                        |> Review.Test.run ruleWithDefaults
+                        |> Review.Test.expectErrors
+                            [ Review.Test.error
+                                { message = "This can be replaced with a call to Set.isEmpty"
+                                , details =
+                                    [ "Set.size takes as long to run as the list is long"
+                                    , "whereas Set.isEmpty runs in constant time."
+                                    ]
+                                , under = "size s == 0"
+                                }
+                                |> Review.Test.whenFixed """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = isEmpty s 
```

Comment:
How do I get the best name for the package as it is imported here? Replacing with `CoreSet.isEmpty` is also fine (preferable, even) but `Set.isEmpty` would be wrong here.

As a real world example, consider a codebase where is used `rtfeldman/elm-sorter-experiment`. Then in a file you would have

```elm
import Sort.Set as Set -- Because you are creating a lot of sets of custom types
import Set as CoreSet -- for the one time you make a set of integers.
```

Reply:
If I'm not mistaken, you'd do that like this:
```elm
qualifiedToString (qualify ( [ "Set" ], "isEmpty" ) checkInfo)
```